app/models/user_job.rb

```diff
@@ -31,6 +31,20 @@ def completion
     "Completed #{completed_at}"
   end
 
+  def complete?
+    completed_at != nil
+  end
+
+  def self.create_and_link_to_user(job_id:, user:, job_title:)
```

Comment:
This method checks before creating so that it works regardless of whether the job is executed immediately or delayed.